_posts/2023-06-23-run-and-debug-rust-verification-harnesses-natively-on-vscode.md

```diff
@@ -0,0 +1,174 @@
+
+#### Run Kani-generated test
+
+Running the unit test using the Run Test (Kani) button, shows us what we’re expecting–that the current unit test is failing.
+This is because the unit test is using the counter-example to invoke the function `stretched_rectangle_can_hold_original`.
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/vs-code-images/run-concrete-playback-test.png" alt="run-concrete-playback-test" />
+
+### Debug Kani unit test
+
+In order to peek under the hood to find out the missing assumptions that lead to unexpected behavior, it is really important to look at the concrete counter examples for which our assertions fail.
+By setting breakpoints and clicking the debug test (Kani) button, you are taken into the debugger which allows you to look at the specific values for which the assertion fails.
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/vs-code-images/run-debugger.gif" style="box-shadow: 0 2px 4px rgba(0, 0, 0, 0.5);" alt="run-debugger" />
+
+In our case, we can see that for `original.height = 0` , the larger rectangle’s height or `larger.height` also stays 0, which shows that for that counter-example, the property `can_hold`  does not hold.
+
+### And finally, verify the harness with the right assumptions
+
+Now that we know that for `original.width = 0`, our assertion fails, we can repeat the experiment with explicit assumptions.  The experiments should reveal that for all parameters, having a 0 value will cause the assertion to fail.
+Additionally, there is a problem if `factor` is `1` because in this case stretch will return `Some(...)` but the stretched rectangle will be the same size as the original.
+We missed these cases in our unit and property-based tests.
+
+We will now add these assumptions through `kani::assume` and re-run the verification in the extension.
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/vs-code-images/verifying-success.gif" style="box-shadow: 0 2px 4px rgba(0, 0, 0, 0.5);" alt="verifying-success" />
+
+And with that green check-mark, you can be assured that the harness has been verified!
+
+
+## Wrapping up
+
+You can use Kani natively in VS Code using the Kani VS Code extension, [available on the marketplace](https://marketplace.visualstudio.com/items?itemName=model-checking.kani-vscode-extension) now.
+We've seen how the VS Code extension can help you to iteratively verify properties of your code.
+The extension can run your Kani harnesses; generate unit tests that demonstrate property violations; and verify the harnesses.
+
+The Kani extension has more features which weren’t mentioned in the blog, that you can read about in our [user guide documentation](https://github.com/model-checking/kani-vscode-extension/blob/main/docs/user-guide.md).
+If you are running into issues with the Kani extension or have feature requests or suggestions, we’d [love to hear from you](https://github.com/model-checking/kani-vscode-extension/issues).
```

Comment:
Let's ask users to try the extension first, then let's talk about what to do if they encounter issues.

Reply:
Also, https://github.com/model-checking/kani-vscode-extension/issues/new/choose brings you to the "New issue" webpage.